### ISSUE TITLE
topic-auto-create option is not working

### ISSUE THREAD

**Issue by tibkiss** [gerritjvv/kafka-fast#45]
With kafka-fast version 3.6.8 the `create-connection` function accepts `:topic-auto-create` flag:
https://github.com/gerritjvv/kafka-fast/blob/master/kafka-clj/src/kafka_clj/client.clj#L292

If the flag is set and the topic to be written to does not exists the `send-msg` function throws error:
```clojure
(def conn (kafka/create-connector [{:host "localhost" :port 9092}]
                               {:flush-on-write true
                                :topic-auto-create true
                                :acks 0}))
(kafka/send-msg conn "non-existent-topic" (.getBytes "Hello"))
```

Exception:
```
RuntimeException No healthy brokers available for topic non-existent-topic 
{:send-cache nil, :state {:shutdown-flag #object[java.util.concurrent.atomic.AtomicBoolean 0x6e6e027a "false"], :producer-error-ch #object[clojure.core.async.impl.channels.ManyToManyChannel 0x73f78c1d "clojure.core.async.impl.channels.ManyToManyChannel@73f78c1d"], :brokers-metadata-ref #object[clojure.lang.Ref 0x4c298c72 {:status :ready, :val {"XYZ" [{:host "192.168.1.101", :port 9092, :isr [{:host "192.168.1.101", :port 9092}], :id 0, :error-code 0}], "ASD" [{:host "192.168.1.101", :port 9092, :isr [{:host "192.168.1.101", :port 9092}], :id 0, :error-code 0}], "check" [{:host "192.168.1.101", :port 9092, :isr [{:host "192.168.1.101", :port 9092}], :id 0, :error-code 0}]}}], :blacklisted-metadata-producers-ref #object[clojure.lang.Ref 0x18b4fb8a {:status :ready, :val {}}], :retry-cache {:db #object[org.mapdb.DB 0x4a490dba "org.mapdb.DB@4a490dba"], :file #object[java.io.File 0x1745551b "/tmp/kafka-retry-cache"], :cache {}}, :metadata-producers-ref #object[clojure.lang.Ref 0x261e911f {:status :ready, :val {{:host "localhost", :port 9092} #object[clojure.lang.Delay 0x517a5dc2 {:status :ready, :val #kafka_clj.produce.Producer{:client #clj_tcp.client.Client{:group #object[io.netty.channel.nio.NioEventLoopGroup 0x3b209536 "io.netty.channel.nio.NioEventLoopGroup@3b209536"], :read-group #object[io.netty.channel.nio.NioEventLoopGroup 0x3b209536 "io.netty.channel.nio.NioEventLoopGroup@3b209536"], :channel-f #object[io.netty.channel.DefaultChannelPromise 0x57cc9e47 "DefaultChannelPromise@57cc9e47(success)"], :write-ch #object[clojure.core.async.impl.channels.ManyToManyChannel 0x64d1be0a "clojure.core.async.impl.channels.ManyToManyChannel@64d1be0a"], :read-ch #object[clojure.core.async.impl.channels.ManyToManyChannel 0x7d878a26 "clojure.core.async.impl.channels.ManyToManyChannel@7d878a26"], :internal-error-ch #object[clojure.core.async.impl.channels.ManyToManyChannel 0x59a577fa "clojure.core.async.impl.channels.ManyToManyChannel@59a577fa"], :error-ch #object[clojure.core.async.impl.channels.ManyToManyChannel 0x6aa8f42f "clojure.core.async.impl.channels.ManyToManyChannel@6aa8f42f"], :reconnect-count 0, :closed #object[java.util.concurrent.atomic.AtomicBoolean 0x360269e "false"], :id #uuid "a99703cb-299d-41b1-a2ca-3a67cd66c47a"}, :host "localhost", :port 9092}}]}}], :scheduled-service #object[java.util.concurrent.Executors$DelegatedScheduledExecutorService 0x1dbb6ccd "java.util.concurrent.Executors$DelegatedScheduledExecutorService@1dbb6ccd"], :activity-counter 0, :conf {:flush-on-write true, :topic-auto-create true, :acks 0, :batch-fail-message-over-limit true, :batch-byte-limit 10485760}, :blacklisted-producers-ref #object[clojure.lang.Ref 0x1feded32 {:status :ready, :val {}}], :producers-cache-ref #object[clojure.lang.Ref 0x67b1709b {:status :ready, :val {}}]}, :msg-ch #object[clojure.core.async.impl.channels.ManyToManyChannel 0x281a1f97 "clojure.core.async.impl.channels.ManyToManyChannel@281a1f97"], :update-metadata #object[kafka_clj.client$create_connector$update_metadata__11932 0x1ef527d2 "kafka_clj.client$create_connector$update_metadata__11932@1ef527d2"], :async-latch #object[java.util.concurrent.CountDownLatch 0x72acb93f "java.util.concurrent.CountDownLatch@72acb93f[Count = 1]"], :persist-delay-thread #object[clojure.core.async.impl.channels.ManyToManyChannel 0xa32f65a "clojure.core.async.impl.channels.ManyToManyChannel@a32f65a"], :retry-cache {:db #object[org.mapdb.DB 0x4a490dba "org.mapdb.DB@4a490dba"], :file #object[java.io.File 0x1745551b "/tmp/kafka-retry-cache"], :cache {}}, :metadata-producers-ref #object[clojure.lang.Ref 0x261e911f {:status :ready, :val {{:host "localhost", :port 9092} #object[clojure.lang.Delay 0x517a5dc2 {:status :ready, :val #kafka_clj.produce.Producer{:client #clj_tcp.client.Client{:group #object[io.netty.channel.nio.NioEventLoopGroup 0x3b209536 "io.netty.channel.nio.NioEventLoopGroup@3b209536"], :read-group #object[io.netty.channel.nio.NioEventLoopGroup 0x3b209536 "io.netty.channel.nio.NioEventLoopGroup@3b209536"], :channel-f #object[io.netty.channel.DefaultChannelPromise 0x57cc9e47 "DefaultChannelPromise@57cc9e47(success)"], :write-ch #object[clojure.core.async.impl.channels.ManyToManyChannel 0x64d1be0a "clojure.core.async.impl.channels.ManyToManyChannel@64d1be0a"], :read-ch #object[clojure.core.async.impl.channels.ManyToManyChannel 0x7d878a26 "clojure.core.async.impl.channels.ManyToManyChannel@7d878a26"], :internal-error-ch #object[clojure.core.async.impl.channels.ManyToManyChannel 0x59a577fa "clojure.core.async.impl.channels.ManyToManyChannel@59a577fa"], :error-ch #object[clojure.core.async.impl.channels.ManyToManyChannel 0x6aa8f42f "clojure.core.async.impl.channels.ManyToManyChannel@6aa8f42f"], :reconnect-count 0, :closed #object[java.util.concurrent.atomic.AtomicBoolean 0x360269e "false"], :id #uuid "a99703cb-299d-41b1-a2ca-3a67cd66c47a"}, :host "localhost", :port 9092}}]}}], :metadata-error-ch #object[clojure.core.async.impl.channels.ManyToManyChannel 0x6ac9577a "clojure.core.async.impl.channels.ManyToManyChannel@6ac9577a"], :flush-on-write true, :producers-cache-ref #object[clojure.lang.Ref 0x67b1709b {:status :ready, :val {}}]}  kafka-clj.client/send-msg (client.clj:69)
```

**Comment by gerritjvv** [comment]
Hi, thanks for reporting this, I'm currently on holiday, will look at it in about a week.

**Comment by gerritjvv** [comment]
fixed in 4.0.0.

**Comment by tibkiss** [comment]
Thanks for the fix, will try to out soon.
